### PR TITLE
[FIX] account: wrong sequence number reset for refund type moves

### PR DESCRIPTION
**Pull request by alt-odoo**
In case a yearly sequence is set for out_invoice/in_invoice types and if we set
a monthly sequence on the corresponding out_refund/in_refund type, we will not be
able to validate the refund the next month as it will be wrongly identified as a
yearly sequence.

Therefore, the constraint on date sequence is preventing to validate it. We should
include the move type when retrieving the last sequence name to solve it.

Description of the issue/feature this PR addresses:
opw-2456008

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -1153,7 +1153,7 @@ def _get_last_sequence_domain(self, relaxed=False):
         param = {'journal_id': self.journal_id.id}
 
         if not relaxed:
-            domain = [('journal_id', '=', self.journal_id.id), ('id', '!=', self.id or self._origin.id), ('name', 'not in', ('/', False))]
+            domain = [('journal_id', '=', self.journal_id.id), ('move_type', '=', self.move_type), ('id', '!=', self.id or self._origin.id), ('name', 'not in', ('/', False))]
```
Comment (william-andre) [inline addons/account/models/account_move.py:1156]: You need to add that new clause iff `self.journal_id.refund_sequence` (like done on line 1168)
This is probably why runbot is red too

**Review by william-andre — CHANGES_REQUESTED** [review]

File: addons/account/models/account_move.py (hunk 1 above)
Comment (alt-odoo) [reply addons/account/models/account_move.py:1156]: I agree with you, I have adapted but the test still fails.

After investigation, I think the test is not correct: 

https://github.com/odoo/odoo/blob/7a7dc212af7279544cffba362bce7daa4fc00865/addons/account/tests/test_reconciliation_matching_rules.py#L744

--> we are posting a move_type 'entry' in a sales journal (NB: `self.journal_id.refund_sequence` will still be `True` ), which does not seem allowed based on some tests made on Runbot.

Should I change the default journal to 'default_journal_misc' instead?

Comment (william-andre) [reply addons/account/models/account_move.py:1156]: They change made should be something like
```python
if self.journal_id.refund_sequence:
    refund_types = ('out_refund', 'in_refund')
    domain+= [('move_type', 'in' if self.move_type in refund_types else 'not in', refund_types)]
```
Do you have something similar?

Comment (william-andre) [reply addons/account/models/account_move.py:1156]: But the test is indeed weird. At first glance what you do here should not break it; but the test is indeed wrong and needs to be changed.

**Review by william-andre — COMMENTED** [review]
Thank you!
@robodoo r+

**Review by william-andre — APPROVED** [review]

**Comment by william-andre** [comment]
@robodoo rebase-ff

**Pull request closed without merging**